pyomo/solvers/plugins/solvers/GAMS.py

```diff
@@ -40,8 +40,24 @@
 )
 
 from pyomo.common.dependencies import attempt_import
+import numpy as np
```

Comment:
This adds a hard dependency on numpy.  Importing numpy from `dependencies` will keep this as a "soft" dependency:
```suggestion
from pyomo.common.dependencies import attempt_import, numpy as np
```

It will still make numpy a hard dependency if you are going to use GAMS, but that is OK (although GAMS's available() should probably check both numpy_available and gdxcc_available)